react/modules/conditions.test.ts

```diff
@@ -0,0 +1,308 @@
+// export function validateConditions({
+//   matchType,
+//   conditions,
+//   values,
+//   handlers,
+// }: {
+//   matchType: MatchType
+//   conditions: Condition[]
+//   values: Record<string, unknown>
+//   handlers: Record<string, unknown>
+// }) {
+
+import { validateConditions } from './conditions'
+import { Condition } from '../types'
+
+const handlers = {
+  potatoId({ values, args }: unknown) {
+    return values.potatoId === args.potatoId
+  },
+  kiwiId({ values, args }: unknown) {
+    return values.kiwiId === args.kiwiId
+  },
+}
+
+describe('matchType: all', () => {
```

Comment:
I think those things could be tested as the user would use. Right now it's testing internal implementation.

What do you think?

Reply:
But would we need to test each match type with every new condition context? I mean, suppose we did these tests in `ConditionLayout.product`. If we came to have other ConditionLayouts, we would need to create the same tests again, right?

Reply:
I don't think it's necessary to add for every context. I would move this to the `ConditionLayout.test.tsx` as it is the generic one.